trie/merkle.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+
+import hashlib
+import json
+import sys
+
+## Reference implementation
+
+def hash(data):
+    return hashlib.blake2b(data, digest_size=32).digest()
+
+def fork(l, r):
+    assert len(l) == 32
+    assert len(r) == 32
+    head = l[0] & 0x7f
+    return bytes([head]) + l[1:] + r
+
+def leaf(k, v):
+    if 0 <= len(v) <= 32:
```

Comment:
Can just drop the `0 <=` here?

Reply:
The spec currently says `0 < len(v) <= 32` but I think it would be simplest for 0-length values to go through the inline-value path, so we should change the spec to match this?

Reply:
Yes, I'll make PR to fix it in the spec